ast-checks/src/no-uri-literals.rs

```diff
@@ -0,0 +1,71 @@
+use codespan::{FileId, Files};
+use common_structs::{NixpkgsHammerMessage, SourceLocation};
+use rnix::{types::*, SyntaxKind::*};
+use std::{collections::HashMap, env, error::Error, fs};
+use tree_utils::walk_kind;
+
+mod common_structs;
+mod tree_utils;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args: Vec<String> = env::args().skip(1).collect();
```

Comment:
Lot of this is same for all checks. Can we factor it out somehow?

Reply:
Yeah, I bet we can!